httpx/dispatch/http2.py

```diff
@@ -139,25 +142,27 @@ def initiate_connection(self) -> None:
             # read timeouts, if we haven't already.
             self.timeout_flags[stream_id].set_read_timeouts()
             if isinstance(event, h2.events.ResponseReceived):
-                break
-
-        status_code = 200
-        headers = []
-        for k, v in event.headers:
-            if k == b":status":
-                status_code = int(v.decode("ascii", errors="ignore"))
-            elif not k.startswith(b":"):
-                headers.append((k, v))
-        return status_code, headers
+                status_code = 200
+                headers = []
+                for k, v in event.headers:
+                    if k == b":status":
+                        status_code = int(v.decode("ascii", errors="ignore"))
+                    elif not k.startswith(b":"):
+                        headers.append((k, v))
+                return (status_code, headers)
 
     async def body_iter(
         self, stream_id: int, timeout: TimeoutConfig = None
     ) -> typing.AsyncIterator[bytes]:
         while True:
             event = await self.receive_event(stream_id, timeout)
             if isinstance(event, h2.events.DataReceived):
-                self.h2_state.acknowledge_received_data(len(event.data), stream_id)
+                self.h2_state.acknowledge_received_data(
+                    event.flow_controlled_length, stream_id
+                )
                 yield event.data
+                if self.h2_state.streams[stream_id].open:
+                    await self.stream.write(self.h2_state.data_to_send(), timeout)
```

Comment:
Why do we still need these 2 lines? As I see it `.receive_event` will process all write operations inside itself, which means that message about increasing the size of the window will be included in the sending payload. So if the stream is open, then the data will definitely be sent there or even earlier. Am I understanding that properly?

Reply:
You are right, these two lines are redundant. I'll remove them.